hyperspy/model.py

```diff
@@ -924,475 +956,698 @@ def red_chisq(self):
             ' reduced chi-squared'
         return tmp
 
-    def fit(self, fitter="leastsq", method='ls', grad=False,
-            bounded=False, ext_bounding=False, update_plot=False,
-            **kwargs):
+    def _calculate_parameter_std(self, pcov, cost, ysize):
+        warn_cov = False
+
+        if pcov is None:  # Indeterminate covariance
+            p_var = np.zeros(len(self.p0), dtype=float)
+            p_var.fill(np.inf)
```

Comment:
This is consistent with how `scipy.optimize.curve_fit` does it. I can replace it with `np.nan` if you prefer, but I think `np.inf` is fine.

Reply:
The things is that in hyperspy several functions (e.g. ``print_summary_statistics``, 2D plotting constrast adjustement...) know how to deal with ``nans`` but not with ``inf``s.

Reply:
OK, I'll replace with nan in that case.

Reply:
Would it be better to fix the functions which doesn't support `np.inf` to keep the consistency with scipy?

Reply:
I don't know how to best do that. Actually, I don't know why ``curve_fit`` uses ``inf`` instead of ``nan`` for this, since a ``nan`` seems to me a better description of the situation. In hyperspy: if a dataset contains ``inf``s, I think that it is up to the user to decide how to best handle them, because the best approach depends on the situation.

Reply:
To be fair I think nan is probably better here after some though - it's not that the parameter estimate is very (i.e. infinitely) large, rather, it cannot be estimated accurately and therefore should not be relied upon.